sql/core/src/main/scala/org/apache/spark/sql/execution/BaseScriptTransformationExec.scala

```diff
@@ -69,9 +75,19 @@ trait BaseScriptTransformationExec extends UnaryExecNode {
     }
   }
 
-  protected def initProc: (OutputStream, Process, InputStream, CircularBuffer) = {
-    val cmd = List("/bin/bash", "-c", script)
+  protected def initProc(hadoopConf: Configuration): ProcParameters = {
+    val wrapper = splitArgs(hadoopConf.get(SQLConf.SCRIPT_TRANSFORMATION_COMMAND_WRAPPER.key))
+    val cmdArgs = splitArgs(script)
+    val prog = cmdArgs(0)
+    if(!new File(prog).isAbsolute) {
+      val progFile = new File(SparkFiles.get(prog))
+      if (progFile.exists()) {
+        cmdArgs(0) = progFile.getAbsolutePath
+      }
+    }
+    val cmd = wrapper.toList ++ cmdArgs.toList
     val builder = new ProcessBuilder(cmd.asJava)
+      .directory(new File(SparkFiles.getRootDirectory()))
```

Comment:
@AngersZhuuuu, is this required? In executor side, the root directory is already current working directory IIRC.

Reply:
> @AngersZhuuuu, is this required? In executor side, the root directory is already current working directory IIRC.

Yea, can remove this

Reply:
> @AngersZhuuuu, is this required? In executor side, the root directory is already current working directory IIRC.

But in Unit Test, driver node's working dir is not `SparkFile.getRootDirectory`. If remove this, test will failed.

Reply:
That seems like because we're running with `local `cluster. If you run the tests with `local-cluster`, I guess it'd work. But okay, in this way we can cover `local` too.

Reply:
Yea,  It seems to have no effect on other tests. Cover local mode maybe better.

Reply:
And add `PathFInder` like hive, it cab support search file under all entry of `PATH`, should be better.

Reply:
`TRANSFORM` does not seem to require an `add file` operation first, which will cause some users to find the file when `TRANSFORM` is running.

Here we can change back to the original behavior, using the workdir of the current process?